ambry-clustermap/src/test/java/com.github.ambry.clustermap/MockHelixCluster.java

```diff
@@ -158,6 +158,11 @@ void setReplicaState(AmbryPartition partition, String instance, TestUtils.Replic
     for (MockHelixAdmin helixAdmin : helixAdmins.values()) {
       upInstances.addAll(helixAdmin.getUpInstances());
     }
+    // add cloud datanodes
+    dataCenterToZkAddress.values()
+        .stream()
+        .filter(info -> info.getReplicaType() == ReplicaType.CLOUD_BACKED)
+        .forEach(info -> upInstances.add(ClusterMapUtils.getInstanceName(info.getDcName(), -1)));
```

Comment:
Suggest using a constant for `-1` and put it into somewhere like TestUtil that can be shared by all tests.